src/capi.rs

```diff
@@ -0,0 +1,109 @@
+use std::boxed::Box;
+use std::os::raw::{c_float, c_int};
+
+use libc::FILE;
+
+pub struct DenoiseState(crate::DenoiseState);
```

Comment:
Why do we need this? I mean, as opposed to just using `*mut crate::DenoiseState` directly?

Reply:
Ideally we can, but this way we do not have cbindgen copying over the rust documentation in the C header and potentially we can have C-specific documentation instead.